src/pages/Browse/index.spec.tsx

```diff
@@ -542,4 +543,199 @@ describe(__filename, () => {
 
     expect(root.find('title')).toHaveText(`Browse ${name}: ${versionString}`);
   });
+
+  it('dispatches fetchVersionFile for the next file', () => {
+    const file1 = 'file1.js';
+    const file2 = 'file2.js';
+    const {
+      _fetchVersionFile,
+      addonId,
+      fakeThunk,
+      store,
+      version,
+    } = setUpVersionFileUpdate({
+      extraFileEntries: {
+        [file1]: { ...fakeVersionEntry, path: file1 },
+        [file2]: { ...fakeVersionEntry, path: file2 },
+      },
+      initialPath: file1,
+    });
+    store.dispatch(
+      fileTreeActions.buildTree({
+        version: createInternalVersion(version),
+      }),
+    );
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    render({
+      _fetchVersionFile,
+      addonId: String(addonId),
+      store,
+      versionId: String(version.id),
+    });
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_fetchVersionFile).toHaveBeenCalledWith({
+      addonId,
+      path: file2,
+      versionId: version.id,
+    });
+  });
+
+  it('does not dispatch fetchVersionFile for the next file if the current version file is not loaded', () => {
+    const file1 = 'file1.js';
+    const file2 = 'file2.js';
+    const {
+      _fetchVersionFile,
+      addonId,
+      store,
+      version,
+    } = setUpVersionFileUpdate({
+      extraFileEntries: {
+        [file1]: { ...fakeVersionEntry, path: file1 },
+        [file2]: { ...fakeVersionEntry, path: file2 },
+      },
+      initialPath: file1,
+      loadVersionAndFile: false,
+    });
+    store.dispatch(versionsActions.loadVersionInfo({ version }));
+    store.dispatch(
+      versionsActions.setCurrentVersionId({ versionId: version.id }),
+    );
+    store.dispatch(
+      fileTreeActions.buildTree({
```

Comment:
`buildTree` and a couple other things get repeated quite a bit. I suggest putting these new tests in their own block, like `describe('preloading', ...)` and introduce a new helper (which could wrap `setUpVersionFileUpdate`) to factor out the repetition. It might get called like this:

```tsx
    const file1 = 'file1.js';
    const file2 = 'file2.js';
    const {
      _fetchVersionFile,
      addonId,
      store,
      version,
    } = setUpFilesForPreload({
      extraFilePaths: [file1, file2],
      initialPath: file1,
    });
```

And when you need to disable certain parts you could do so with flags:

```tsx
    const file1 = 'file1.js';
    const file2 = 'file2.js';
    const {
      _fetchVersionFile,
      addonId,
      store,
      version,
    } = setUpFilesForPreload({
      extraFilePaths: [file1, file2],
      initialPath: file1,
      buildFileTree: false,
    });
```

Reply:
Great. New helper makes things much easier.